### PR TITLE
feat: configure ipv4 addon

### PR DESCRIPTION
**Pull request by kevcodez**
Can only be merged with the next prod API deployment

### CODE REVIEW

File: apps/studio/components/interfaces/Settings/Addons/Addons.tsx [hunk 1]
```diff
@@ -343,6 +345,84 @@ const Addons = () => {
...
+                </div>
+              </ScaffoldSectionDetail>
+              <ScaffoldSectionContent>
+                <Alert_Shadcn_ variant="warning">
+                  <IconAlertTriangle className="h-4 w-4" />
+                  <AlertTitle_Shadcn_>
+                    PGBouncer and IPv4 Deprecation on January, 26th
+                  </AlertTitle_Shadcn_>
+                  <AlertDescription_Shadcn_>
+                    IPv4 for direct connections or through db.projectref.supabase.co will resolve to
```
Comment (inian) [inline apps/studio/components/interfaces/Settings/Addons/Addons.tsx:377]: Direct connections via db.projectref.supabase.co will resolve to an IPv6 address starting from January 26th.

File: apps/studio/components/interfaces/Settings/Addons/Addons.tsx [hunk 2]
```diff
@@ -343,6 +345,84 @@ const Addons = () => {
...
+                    an IPv6 address starting from January 26th. If you plan on not using our
+                    connection pooler and your environment does not support IPv4, consider enabling
```
Comment (inian) [inline apps/studio/components/interfaces/Settings/Addons/Addons.tsx:379]: and your environment does not support IPv6

File: apps/studio/components/interfaces/Settings/Addons/IPv4SidePanel.tsx [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+              About IPv4
+            </Link>
+          </Button>
+        </div>
+      }
+    >
+      <SidePanel.Content>
+        <div className="py-6 space-y-4">
+          <p className="text-sm">
+            By default direct connections to the database only work via IpV6. Enabling IPv4 allows
```
Comment (inian) [inline apps/studio/components/interfaces/Settings/Addons/IPv4SidePanel.tsx:157]: Direct connections to the database only work if your client is able to resolve IPv6 addresses from January 26th. Enabling the IPv4 add-on allows you to directly connect to your database via a IPv4 address. If you are connecting via our connection Pooler, you do not need this add-on as our pooler supports IPv4 addresses. You can check your connection info...

File: apps/studio/components/interfaces/Settings/Addons/IPv4SidePanel.tsx [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+                )
+              ) : (
+                <p className="text-sm text-foreground-light">
+                  Upon clicking confirm, the amount of{' '}
+                  <span className="text-foreground">${selectedIPv4?.price.toLocaleString()}</span>{' '}
+                  will be added to your monthly invoice.
+                </p>
+              )}
+            </>
+          )}
```
Comment (ecktoteckto) [inline apps/studio/components/interfaces/Settings/Addons/IPv4SidePanel.tsx:260]: **(note):** I'm pretty sure it's only me who's a bit confused here. But maybe you can double check whether the texts are correct (regarding "Prepaid", "End-of-Month", "Usage-Based", "Prorated Credits" etc.).

File: apps/studio/components/interfaces/Settings/Addons/IPv4SidePanel.tsx [hunk 5]
```diff
@@ -0,0 +1,281 @@
...
+          </Button>
+        </div>
+      }
+    >
+      <SidePanel.Content>
+        <div className="py-6 space-y-4">
+          <p className="text-sm">
+            Direct connections to the database only work if your client is able to resolve IPv6
+            addresses from January 26th. Enabling the IPv4 add-on allows you to directly connect to
+            your database via a IPv4 address. If you are connecting via our connection Pooler, you
```
Comment (joshenlim) [inline apps/studio/components/interfaces/Settings/Addons/IPv4SidePanel.tsx:159]:
```suggestion
            your database via a IPv4 address. If you are connecting via our connection pooler, you
```

File: apps/studio/components/interfaces/Settings/Addons/IPv4SidePanel.tsx [hunk 6]
```diff
@@ -0,0 +1,281 @@
...
+            do not need this add-on as our pooler resolves to IPv4 addresses. You can check your
+            connection info in your{' '}
+            <Link href={`/project/${projectRef}/settings/database`} className="text-brand">
+              Project database Settings
```
Comment (joshenlim) [inline apps/studio/components/interfaces/Settings/Addons/IPv4SidePanel.tsx:163]:
```suggestion
              project database settings
```

File: apps/studio/components/interfaces/Settings/Addons/IPv4SidePanel.tsx [hunk 7]
```diff
@@ -0,0 +1,281 @@
...
+      header={
+        <div className="flex items-center justify-between">
+          <h4>IPv4</h4>
+          <Button asChild type="default" icon={<IconExternalLink strokeWidth={1.5} />}>
+            <Link
+              href="https://github.com/orgs/supabase/discussions/17817"
+              target="_blank"
+              rel="noreferrer"
+            >
+              About IPv4
```
Comment (joshenlim) [inline apps/studio/components/interfaces/Settings/Addons/IPv4SidePanel.tsx:148]: the CTA sounds weird here

the link is to the deprecation notice but this "About IPv4" sounds like a doc about IPv4 address?
```suggestion
              About IPv4 deprecation
```

File: apps/studio/components/interfaces/Settings/Addons/Addons.tsx [hunk 8]
```diff
@@ -343,6 +345,83 @@ const Addons = () => {
...
+                <Alert_Shadcn_ variant="warning">
+                  <IconAlertTriangle className="h-4 w-4" />
+                  <AlertTitle_Shadcn_>
+                    PGBouncer and IPv4 Deprecation on January, 26th
+                  </AlertTitle_Shadcn_>
+                  <AlertDescription_Shadcn_>
+                    Direct connections via db.{projectRef}.supabase.co will resolve to an IPv6
+                    address starting from January 26th. If you plan on not using our connection
+                    pooler and your environment does not support IPv6, consider enabling the add-on.
+                  </AlertDescription_Shadcn_>
```
Comment (joshenlim) [inline apps/studio/components/interfaces/Settings/Addons/Addons.tsx:380]: wondering if we should add a learn more CTA here like the one in database settings

image

File: apps/studio/components/interfaces/Settings/Addons/IPv4SidePanel.tsx (hunk 7 above)
Comment (kevcodez) [reply apps/studio/components/interfaces/Settings/Addons/IPv4SidePanel.tsx:148]: This is just a temporary link until we moved it to docs, adjusted the copy for now

File: apps/studio/components/interfaces/Settings/Addons/Addons.tsx (hunk 8 above)
Comment (kevcodez) [reply apps/studio/components/interfaces/Settings/Addons/Addons.tsx:380]: good one

**Review by joshenlim — APPROVED** [review]

**Pull request merged**